src/manager.cpp

```diff
@@ -212,12 +212,93 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    if (i_vpdPath.empty())
+    {
+        logging::logMessage("Given VPD path is empty.");
+        return -1;
+    }
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    nlohmann::json l_sysCfgJsonObj{};
+
+    if (m_worker.get() != nullptr)
+    {
+        l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    int l_bytesUpdated = -1;
+    types::Path l_hwPath = i_vpdPath;
+    types::Path l_redundantFruPath;
+
+    try
+    {
```

Comment:
move this try catch for if JSON available only

Reply:
and for update on primary hardware and redundant hardware, you want to have a separate try catch?

Reply:
This try catch is only for JSON based check so better too move it under the check of JSON object

Reply:
I do not see that there is a need for try catch for the hardware calls as per this PR so asked this change

Reply:
done